iceoryx2/tests/subscriber_tests.rs

```diff
@@ -69,6 +71,31 @@ mod subscriber {
         }
     }
 
+    #[test]
+    #[should_panic]
+    #[cfg(debug_assertions)]
+    fn subscriber_with_custom_payload_details_panics_when_calling_receive<Sut: Service>() {
```

Comment:
```suggestion
    fn subscriber_with_custom_payload_details_panics_when_calling_non_custom_receive<Sut: Service>() {
```

What do you think? I was a bit confused by the name at the beginning, since I thought a condition was missing but it's related to call `receive` instead of `receive_custom_payload`. If you agree, `publisher_with_custom_payload_details_panics_when_calling_loan_slice_uninit` could also be adjusted.